src/Uno.UI/UI/Xaml/Controls/Canvas/Canvas.Android.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Windows.UI.Xaml.Controls
+{
+	public partial class Canvas
+	{
+		/// <summary>
+		/// Order of children determined by Canvas.ZIndex, keyed by position in Children
+		/// </summary>
+		private Dictionary<int, int> _zSortedChildren;
+
+		partial void InitializePartial()
+		{
+			// Set this in order for GetChildDrawingOrder() to be called
+			ChildrenDrawingOrderEnabled = true;
+		}
+
+		partial void MeasureOverridePartial()
+		{
+			_zSortedChildren = Children
```

Comment:
I'm wondering about the performance of this query, as it is re-creating a dictionary every time. Setting the Z index forcibly creates a shadow ?

Reply:
Setting the native `Z` value forcibly creates a shadow. 

I can rewrite it to reuse the dictionary.

Reply:
Indeed it doesn't need to be a dictionary at all

Reply:
Using an array now